packages/devtools_app/lib/src/shared/config_specific/framework_initialize/_framework_initialize_web.dart

```diff
@@ -15,7 +17,7 @@ import '../../server_api_client.dart';
 /// Return the url the application is launched from.
 Future<String> initializePlatform() async {
   // Clear out the unneeded HTML from index.html.
-  for (var element in document.body!.querySelectorAll('.legacy-dart')) {
+  for (var element in html.document.body!.querySelectorAll('.legacy-dart')) {
```

Comment:
Is this kept as `dart:html` because `NodeList` is not a `List`?

Reply:
Yep:
```
The type 'NodeList' used in the 'for' loop must implement 'Iterable'.dart[for_in_of_invalid_type](https://dart.dev/diagnostics/for_in_of_invalid_type)
```

Reply:
I think long-term we're still deciding the right way to handle list-like JS interfaces. One option would be a conversion function `toDartListLike` that would allow users to treat these interfaces like a `List` with `external` accesses.

For now though, we can use interop in this file to iterate:

```dart
extension NodeListExtension on NodeList {
  external void forEach(JSFunction callback);
}

...

document.body!.querySelectorAll('.legacy-dart').forEach((Node element) {
  // This should be the equivalent of `element.remove()` from `dart:html`.
  if (element.parent != null) {
    element.parent!.removeChild(element);
  }
}.toJS);

```

There's a bug where `forEach` is not emitted on `iterable` types so that's why there's a separate definition here: https://github.com/dart-lang/web/issues/21

Reply:
I think `Node` is the wrong type here. Do I need to cast `element` to something else for this to work?
![Screenshot 2023-10-31 at 3 05 33 PM](https://github.com/flutter/devtools/assets/43759233/601246b4-fceb-43aa-b2bc-cfb38df64a5c)

Reply:
Egg on my face for not validating that API exists first. I believe the right API is `parentNode`, not `parent`. :)